### PR TITLE
Add `.nonEmptyStringAndNotWhitespace()`

### PR DESCRIPTION
**Pull request by ivankatliarchuk**
as described here https://github.com/sindresorhus/is/issues/158

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -245,6 +245,17 @@ Returns `true` if the value is a `string` and the `.length` is 0.
 
 Returns `true` if the value is a `string` and the `.length` is more than 0.
 
+##### .nonEmptyStringAndNotWhitespace(value)
+
+Returns `true` if `is.string(value)` and if it's a `string` that is not a whitespace.
```
Comment (sindresorhus) [inline readme.md:250]: This description needs work. "not a whitespace" is correct. And it's also missing "non-empty".

File: readme.md [hunk 2]
```diff
@@ -245,6 +245,17 @@ Returns `true` if the value is a `string` and the `.length` is 0.
...
+
+```js
+const values = ['property1', '', null, 'property2', '    ', undefined, , ,]
```
Comment (sindresorhus) [inline readme.md:253]: The array holes are not relevant.
```suggestion
const values = ['property1', '', null, 'property2', '    ', undefined]
```

File: readme.md [hunk 3]
```diff
@@ -245,6 +245,17 @@ Returns `true` if the value is a `string` and the `.length` is 0.
...
+
+values.filter(is.nonEmptyStringAndNotWhitespace)
+//=> (2) ['property1', 'property2']
```
Comment (sindresorhus) [inline readme.md:256]:
```suggestion
//=> ['property1', 'property2']
```

File: test/test.ts [hunk 4]
```diff
@@ -1436,6 +1436,22 @@ test('is.emptyStringOrWhitespace', t => {
 	});
 });
 
+test('is.nonEmptyStringAndNotWhitespace', t => {
+	t.false(is.nonEmptyStringAndNotWhitespace(' '));
+	t.false(is.nonEmptyStringAndNotWhitespace(String()));
+	t.true(is.nonEmptyStringAndNotWhitespace('🦄'));
```
Comment (sindresorhus) [inline test/test.ts:1442]: You should also test some non-string values

File: test/test.ts [hunk 5]
```diff
@@ -1436,6 +1436,22 @@ test('is.emptyStringOrWhitespace', t => {
...
+
+	t.throws(() => {
+		assert.nonEmptyStringAndNotWhitespace('');
+	});
+	t.throws(() => {
+		assert.nonEmptyStringAndNotWhitespace(String());
+	});
```
Comment (sindresorhus) [inline test/test.ts:1449]: This test is not useful.

File: test/test.ts [hunk 6]
```diff
@@ -1436,6 +1436,22 @@ test('is.emptyStringOrWhitespace', t => {
 	});
 });
 
+test('is.nonEmptyStringAndNotWhitespace', t => {
+	t.false(is.nonEmptyStringAndNotWhitespace(' '));
+	t.false(is.nonEmptyStringAndNotWhitespace(String()));
```
Comment (sindresorhus) [inline test/test.ts:1441]: This test is not useful.

File: readme.md (hunk 1 above)
Comment (ivankatliarchuk) [reply readme.md]: updated

File: test/test.ts (hunk 6 above)
Comment (ivankatliarchuk) [reply test/test.ts]: removed

File: test/test.ts (hunk 4 above)
Comment (ivankatliarchuk) [reply test/test.ts:1441]: Wrong assumption. All code has bugs, I actually expect it to be buggy, and someone can catch an edge case and fix it ;-)

Comment (ivankatliarchuk) [reply test/test.ts:1441]: Added better test coverage
